### PR TITLE
feat(api) : changes in permission module

### PR DESCRIPTION
**Pull request by soumyadip007**
Subject:  feat(api) : changes in permission module
Assignees: @soumyadip007 

## Closes / Fixes 

1. Changes in Create Permission Request Modules (changes in activity stream)
2. Changes in Get permission by Action
3. Lint issues resolved and Code formatted

## Does this PR introduce a breaking change

NO

## Screenshot(s)

<details>
<summary>View Screenshots</summary>

</details>

### Ready-for-merge Checklist

- [ ] Expected files: all files in this pull request are related to one feature request or issue (no stragglers)?
- [ ] Does the change have appropriate unit tests?
- [ ] Did you update or add any necessary documentation (README.md, WHY.md, etc.)?
- [ ] Was this feature demo'd and the design review approved?

### CODE REVIEW

File: packages/api/src/server/analytics/service/analytics.service.ts [hunk 1]
```diff
@@ -57,7 +57,7 @@ export class AnalyticsService {
     const savedAnalytics = await this.dataServices.activityStream.create(activityStream);
     try {
       const webhooks = await this.dataServices.webhook.getByAny({ propertyIdentifier, actions: action });
-      this.publishWebhookEvents(webhooks, activityStream);
+      if (webhooks.length !== 0) this.publishWebhookEvents(webhooks, activityStream);
```
Comment (arkaprovob) [inline packages/api/src/server/analytics/service/analytics.service.ts:60]: Can you please raise a different PR for this? Rest of the changes looks good to me.

Comment (arkaprovob) [reply packages/api/src/server/analytics/service/analytics.service.ts:60]: also, remove this from  the description "Provide Initial access sync issue resolved for webhook"

Comment (soumyadip007) [reply packages/api/src/server/analytics/service/analytics.service.ts:60]: Sure @arkaprovob

Comment (soumyadip007) [reply packages/api/src/server/analytics/service/analytics.service.ts]: Hi @arkaprovob , I've removed this.

**Review by arkaprovob — APPROVED** [review]

File: packages/api/src/server/analytics/service/analytics.service.ts (hunk 1 above)
Comment (arkaprovob) [reply packages/api/src/server/analytics/service/analytics.service.ts]: done

**Pull request merged**